pallets/credit/src/lib.rs

```diff
@@ -478,6 +490,36 @@ pub mod pallet {
             Self::update_credit_by_burn_nft(sender.clone(), credit)?;
 
             Self::deposit_event(Event::BurnNft(sender, class_id, instance_id, credit));
+
+            Ok(().into())
+        }
+
+        #[pallet::weight(<T as pallet::Config>::WeightInfo::set_switch_campaign())]
+        pub fn set_switch_campaign(
+            origin: OriginFor<T>,
+            old_ids: Vec<CampaignId>,
+            new_ids: Vec<CampaignId>,
+        ) -> DispatchResultWithPostInfo {
+            ensure_root(origin)?;
+            ensure!(
+                old_ids.len() == new_ids.len(),
+                Error::<T>::CampaignIdNotMatch
+            );
+            for i in 0..old_ids.len() {
+                CampaignIdSwitch::<T>::insert(old_ids[i], new_ids[i]);
+            }
+            Ok(().into())
+        }
+
+        #[pallet::weight(<T as pallet::Config>::WeightInfo::set_switch_accounts())]
+        pub fn set_switch_accounts(
+            origin: OriginFor<T>,
+            accounts: Vec<T::AccountId>,
+        ) -> DispatchResultWithPostInfo {
+            ensure_root(origin)?;
+            for id in accounts {
```

Comment:
Do we need to check if the account already contains in `SwitchAccounts`?

Reply:
no need for storagemap